unskript-ctl/unskript_ctl_config_parser.py

```diff
@@ -0,0 +1,229 @@
+JOB_CONFIG_NOTIFY_KEY_NAME = "notify"
+
+# Credential section related
+CREDENTIAL_CONFIG_SKIP_VALUE_FOR_ARGUMENTS = ["no-verify-certs", "no-verify-ssl", "use-ssl"]
+
+class Job():
+    def __init__(
+            self,
+            checks: list[str],
+            suites: list[str]=None,
+            connectors: list[str] = None,
+            custom_scripts: list[str] = None,
+            notify: bool = False):
+        self.checks = checks
+        self.suites = suites
+        self.connectors = connectors
+        self.custom_scripts = custom_scripts
+        self.notify = notify
+
+    def parse(self):
+        cmds = []
+        notify = '--report' if self.notify == True else ''
+        #TBD: Add support for suites and custom_scripts
+        if self.checks is not None:
+            cmds.append(f'{UNSKRIPT_CTL_BINARY} -rc --check {self.checks[0]} {notify}')
+        if self.connectors is not None:
+            # Need to construct the unskript-ctl command like
+            # unskript-ctl.sh -rc --types aws,k8s
+            connector_types_string = ','.join(self.connectors)
+            cmds.append(f'{UNSKRIPT_CTL_BINARY} -rc --type {connector_types_string} {notify}')
+        self.cmds = cmds
+        print(self.cmds)
+
+
+class ConfigParser():
+    def __init__(self, config_file: str):
+        self.config_file = config_file
+        # Dictionary of jobs, with job name being the key.
+        self.jobs = {}
+
+    def parse_config_yaml(self) -> dict:
+        """parse_config_yaml: This function parses the config yaml file and converts the
+        content as a python dictionary and returns back to the caller.
+        """
+        retval = {}
+
+        if os.path.exists(self.config_file) == False:
+            print(f"WARNING: {self.config_file} Not found!")
+            return retval
+
+
+        # We use EnvYAML to parse the hook file and give us the
+        # dictionary representation of the YAML file
+        retval = EnvYAML(self.config_file, strict=False)
+
+        if not retval:
+            print(f"WARNING: config file {self.config_file} content seems to be empty!")
+            return retval
+
+        self.parsed_config = retval
+
+    def configure_credential(self):
+        """configure_credential: This function is used to parse through the creds_dict and
+        call the add_creds.sh method to populate the respective credential json
+        """
+        print('Processing credential section')
+        creds_dict = self.parsed_config.get('credential')
+        if creds_dict is None:
+            print(f"ERROR: Nothing to configure credential with, found empty creds data")
+            return
+
+        for cred_type in creds_dict.keys():
+            cred_list = creds_dict.get(cred_type)
+            for cred in cred_list:
+                if cred.get('enable') == False:
+                    continue
+                creds_cmd = ['/usr/local/bin/add_creds.sh', '-c', cred_type]
```

Comment:
why not just call the python routine? you already have the add_creds.py, from add_creds import * and call the respective routines? Just a suggestion.

Reply:
yes i thought about that, we can do it as a followup